spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/support/SimpleJpaRepository.java

```diff
@@ -184,7 +184,8 @@ public void delete(T entity) {
 
 		Class<?> type = ProxyUtils.getUserClass(entity);
 
-		T existing = (T) em.find(type, entityInformation.getId(entity));
+		Object id = em.getEntityManagerFactory().getPersistenceUnitUtil().getIdentifier(entity);
```

Comment:
Is that a left-over from initial experiments? We should be able to still call `entityInformation.getId(…)` right?

Reply:
Yes it is. Thanks!